docs/docs/guides/build/components/building-pipelines-with-components/using-template-variables.md

```diff
@@ -0,0 +1,87 @@
+---
+description: Use template variables to inject dynamic values and functions into your component definitions.
+sidebar_position: 300
+title: Using template variables
+---
+
+import DgComponentsPreview from '@site/docs/partials/\_DgComponentsPreview.md';
+
+<DgComponentsPreview />
+
+Template variables provide a powerful way to inject dynamic values and functions into your component definitions. Instead of creating component subclasses just to add template scope, you can use the `@template_var` decorator to define variables that are available in your component templates.
+
+There are two ways to define template variables:
+
+1. **Template variables module** - Defined as standalone functions in a separate module
+2. **Component class static methods** - Defined as `@staticmethod` on a Component class
+
+
+## Template variables module
+
+The template variables module allows for defining functions in a separate module and are associated with a specific path in the component hierarchy. They can optionally accept a `ComponentLoadContext` parameter, giving you access to the component's loading context.
+
+### Setting up a template variables module
+
+First, create a `template_vars.py` file adjacent to your `defs.yaml`:
+
+<CodeExample
+  path="docs_snippets/docs_snippets/guides/components/using-template-variables/template_vars.py"
+  language="python"
+  title="template_vars.py"
+/>
+
+Then reference the template variables module in your `defs.yaml` using Jinja template syntax:
+
+<CodeExample
+  path="docs_snippets/docs_snippets/guides/components/using-template-variables/defs.yaml"
+  language="yaml"
+  title="defs.yaml"
+/>
+
+
+### Module path resolution
+
+The `template_vars_module` field supports both relative and absolute module paths:
+
+- **Relative paths** (starting with `.`) are resolved relative to the current component's module path
+- **Absolute paths** specify the full module path
+
+```yaml
+# Relative path (recommended for co-located template vars)
+template_vars_module: .template_vars
+
+# Absolute path
+template_vars_module: my_project.shared.template_vars
+```
+
+## Template variable function signatures
+
+Template variable functions can have different signatures depending on whether they need access to the component loading context:
+
+<CodeExample
+  path="docs_snippets/docs_snippets/guides/components/using-template-variables/template_var_context.py"
+  language="python"
+  title="template_vars.py"
+/>
+
+
+## Component class static methods
+
+Template variables can be defined directly on your Component class using the `@template_var` decorator on a `@staticmethod`. These variables are associated with the component class itself.
```

Comment:
maybe a bit more explanation for why you would use this method over the separate module approach. My understanding is basically "This approach is useful in cases where you are defining a reusable component and would like users of that component to have access to a set of useful template variables"